internal/resources/oncall/resource_integration.go

```
@@ -792,3 +814,36 @@ func expandDefaultRoute(input []interface{}) *onCallAPI.DefaultRoute {
 	}
 	return &defaultRoute
 }
+
+func expandLabels(input []interface{}) []*onCallAPI.Label {
+	labelsData := make([]*onCallAPI.Label, 0, 1)
+
+	for _, r := range input {
+		inputMap := r.(map[string]interface{})
+		key, keyExists := inputMap["key"]
```

This is a work around, of sorts.

We're using TypeMap here and the preferred way for users to interact with labels as part of oncall integrations is to create a data source object which enforces requiring `key` and `value` as attributes, then passing that object into the list.

A user could, instead, create the object directly within the list such as:
```
...
labels = [
    {
      key = "SomeKey"
      value = "SomeValue"
    }
]
...
```

which would work. That said, `schema.TypeMap` can't enforce required values (see: https://developer.hashicorp.com/terraform/plugin/sdkv2/schemas/schema-types#typemap) and if the user leaves one out, it would normally panic.

This is at least a check so it doesn't totally fall over, but it isn't great UX as it doesn't indicate to the user they've left anything out. We could rework the returns on the `flattenLabels` and other checks such that we can return an error in the `diag` object.